docs/content/v2025.1/explore/transactions/transactional-ddl.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Transactional DDL
+headerTitle: Transactional DDL
+linkTitle: Transactional DDL
+description: Learn how YugabyteDB handles DDLs in a transaction
+headcontent: Learn how YugabyteDB handles DDLs in a transaction
+tags:
+   feature: tech-preview
+menu:
+  v2025.1:
+    identifier: transactional-ddl
+    parent: explore-transactions
+    weight: 300
+type: docs
+---
+
+Data Definition Language (DDL) statements modify the structure of your data. PostgreSQL allows executing such statements within a transactional block and supports rolling them back as part of a transaction rollback. For two transactions undergoing conflicting DDLs, it also provides isolation guarantees by taking appropriate object locks during DDL processing.
+
+YugabyteDB's transactional DDL provides similar guarantees for rolling back DDL operations done inside a transaction block. However, because object locks are still under active development, isolation guarantees are weaker than in PostgreSQL. See [Limitations](#limitations) for more information.
+
+## Enable transactional DDL
+
+To enable transactional DDL, set the [yb-tserver](../../../reference/configuration/yb-tserver/) flag `yb_ddl_transaction_block_enabled` to true.
+
+Because `yb_ddl_transaction_block_enabled` is a preview flag, to use it, add the flag to the [allowed_preview_flags_csv](../../../reference/configuration/yb-tserver/#allowed-preview-flags-csv) list (that is, `allowed_preview_flags_csv=yb_ddl_transaction_block_enabled`).
+
+### Rollback capabilities
+
+All DDLs supported in YugabyteDB provide the same rollback capabilities as PostgreSQL. These include DDLs on tables, indexes, roles, and materialized views.
+
+Note that some DDL statements such as DDLs on database or tablespaces are disallowed in a transaction block in PostgreSQL, and are also disallowed in YugabyteDB.
+
+#### Rollback CREATE TABLE
+
+This example demonstrates rolling back a `CREATE TABLE` statement within a transaction block.
+
+```sql
+yugabyte=# BEGIN;
+yugabyte=*# CREATE TABLE foo (bar int);
+yugabyte=*# INSERT INTO foo VALUES (1);
+yugabyte=*# ROLLBACK;
+yugabyte=# SELECT * FROM foo;
+```
+
+```output
+ERROR:  relation "foo" does not exist
+LINE 1: SELECT * FROM foo;
+                      ^
+```
+
+#### Rollback ALTER TABLE
+
+This example demonstrates rolling back an `ALTER TABLE` statement that adds a column.
+
+```sql
+yugabyte=# CREATE TABLE foo (bar int);
+yugabyte=# BEGIN;
+yugabyte=*# ALTER TABLE foo ADD COLUMN name text;
+yugabyte=*# SELECT * FROM foo;
+```
+
+```output
+ bar | name
+-----+------
+(0 rows)
+```
+
+```sql
+yugabyte=*# ROLLBACK;
+yugabyte=# SELECT * FROM foo;
+ bar
+-----
+(0 rows)
+```
+
+#### Rollback Materialized View
```

Comment:
I'm not sure if this many examples are needed to explain this - I think people understand what rolling back a DDL means. Ideally we can include just one example with DDL, DML together to show the capability.